File: CleverTapSDK/CTKnownProfileFields.m
#import "CTKnownProfileFields.h"
#import "CTConstants.h"

@implementation CTKnownProfileFields {
    
}

+ (KnownField)getKnownFieldIfPossibleForKey:(NSString *)key {
    if ([key isEqualToString:@"Name"])
        return Name;
    else if ([key isEqualToString:@"Email"])
        return Email;
    else if ([key isEqualToString:@"Education"])
        return Education;
    else if ([key isEqualToString:@"Married"])
        return Married;
    else if ([key isEqualToString:@"DOB"])
        return DOB;
    else if ([key isEqualToString:@"Birthday"])
        return Birthday;
    else if ([key isEqualToString:@"Employed"])
        return Employed;
    else if ([key isEqualToString:@"Gender"])
        return Gender;
    else if ([key isEqualToString:@"Phone"])
        return Phone;
    else if ([key isEqualToString:@"Age"])
        return Age;
    else
        return UNKNOWN;
}

@end

File: CleverTapSDK/CTKnownProfileFields.swift
//
//  CTKnownProfileFields.swift
//  CleverTapSDK
//

import Foundation

@objc
public enum KnownField: Int {
    case name = 100
    case email
    case education
    case married
    case dob
    case birthday
    case employed
    case gender
    case phone
    case age
    case unknown
}

@objc(CTKnownProfileFields)
@objcMembers
public class CTKnownProfileFields: NSObject {

    private static let fieldMap: [String: KnownField] = [
        "Name": .name,
        "Email": .email,
        "Education": .education,
        "Married": .married,
        "DOB": .dob,
        "Birthday": .birthday,
        "Employed": .employed,
        "Gender": .gender,
        "Phone": .phone,
        "Age": .age
    ]

    @objc
    public static func getKnownFieldIfPossible(forKey key: String) -> KnownField {
        return fieldMap[key] ?? .unknown
    }
}
